### PR TITLE
fix auto_pipeline: pass kwargs to load_config

### PR DESCRIPTION
**Pull request by yiyixuxu**
This PR fix https://github.com/huggingface/diffusers/issues/4727

We now pass relevant kwargs to `load_config()` when we retrieve the config file 

I only updated for Text2Image here, will update other auto pipelines too if it's ok

### CODE REVIEW

File: src/diffusers/pipelines/auto_pipeline.py [hunk 1]
```diff
@@ -295,15 +296,37 @@ def from_pretrained(cls, pretrained_model_or_path, **kwargs):
...
+        config = cls.load_config(pretrained_model_or_path, **load_config_kwargs)
         orig_class_name = config["_class_name"]
 
         if "controlnet" in kwargs:
             orig_class_name = config["_class_name"].replace("Pipeline", "ControlNetPipeline")
 
         text_2_image_cls = _get_task_class(AUTO_TEXT2IMAGE_PIPELINES_MAPPING, orig_class_name)
 
-        return text_2_image_cls.from_pretrained(pretrained_model_or_path, **kwargs)
+        return text_2_image_cls.from_pretrained(pretrained_model_or_path, **load_config_kwargs, **kwargs)
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/auto_pipeline.py:329]:
```suggestion
        kwargs = {**load_config_kwargs, **kwargs}
        return text_2_image_cls.from_pretrained(pretrained_model_or_path, **kwargs)
```

**Review by patrickvonplaten — COMMENTED** [review]
Can we apply this change also to all other autopipelines to make sure they are also fixed?

**Comment by patrickvonplaten** [comment]
Super, sorry and a small test to check that everything works as expected would be nice (e.g. maybe just passing the `cache_dir` to a dummy model and make sure that both config and model are saved in the same cache dir

**Comment by yiyixuxu** [comment]
@patrickvonplaten 

I think it will be pretty tricky to test, `load_config()` is called twice here: 

first time to find out the original class name for the pipeline, and then it will be called again once we dispatch to the `pipeline_class.from_pretrained()` - so the config and model will always be saved in the cache_dir  the second time we call `load_config`.

**Comment by patrickvonplaten** [comment]
Hmm I think it shouldn't be that difficult to test actually - you could more or less use the reproducible code snippet here: https://github.com/huggingface/diffusers/issues/4727#issue-1862378356 as a way of testing no?

**Comment by patrickvonplaten** [comment]
Hmm yeah sorry this:
> Super! Sorry, one last thing: Let's maybe add a small test to check that everything works as expected would be nice - e.g. maybe just passing the cache_dir to a dummy model and make sure that both config and model are saved in the same cache dir
 
wasn't very clear / didn't make too much sense :sweat_smile: 

Then let's maybe more or less use the reproducible code snippet here: https://github.com/huggingface/diffusers/issues/4727#issue-1862378356 as a way of testing no? Using `only_local_files` as a way of testing should be easy to set up :-)

Maybe you can write a test similar to this one: https://github.com/huggingface/diffusers/blob/c4d282360184686f7d4a66787a4be9898cf1b8b0/tests/pipelines/test_pipelines.py#L441 (you don't need to add the mock internet part though)

**Comment by yiyixuxu** [comment]
@patrickvonplaten 

So the code snippet here https://github.com/huggingface/diffusers/issues/4727#issue-1862378356 isn't reproducible as it is: I think they happened to have an outdated copy of the cached model. 

I made a test to try to recreate this edge case by modifying the commit ID. Let me know if this is ok, or if there is an easier/better way to test.

**Comment by patrickvonplaten** [comment]
Looks great - thanks!

**Review by patrickvonplaten — APPROVED** [review]
